samples/snippets/src/main/java/com/example/bigquerystorage/WritePendingStream.java

```diff
@@ -48,63 +60,124 @@ public static void runWritePendingStream()
-        for (int i = 0; i < 2; i++) {
-          // Create a JSON object that is compatible with the table schema.
-          JSONArray jsonArr = new JSONArray();
-          for (int j = 0; j < 10; j++) {
-            JSONObject record = new JSONObject();
-            record.put("col1", String.format("batch-record %03d-%03d", i, j));
-            jsonArr.put(record);
-          }
-          ApiFuture<AppendRowsResponse> future = writer.append(jsonArr);
-          AppendRowsResponse response = future.get();
-        }
-        FinalizeWriteStreamResponse finalizeResponse =
-            client.finalizeWriteStream(writeStream.getName());
-        System.out.println("Rows written: " + finalizeResponse.getRowCount());
-      }
+    BigQueryWriteClient client = BigQueryWriteClient.create();
+    TableName parentTable = TableName.of(projectId, datasetName, tableName);
 
-      // Commit the streams.
-      BatchCommitWriteStreamsRequest commitRequest =
-          BatchCommitWriteStreamsRequest.newBuilder()
-              .setParent(parentTable.toString())
-              .addWriteStreams(writeStream.getName())
-              .build();
-      BatchCommitWriteStreamsResponse commitResponse =
-          client.batchCommitWriteStreams(commitRequest);
-      // If the response does not have a commit time, it means the commit operation failed.
-      if (commitResponse.hasCommitTime() == false) {
-        for (StorageError err : commitResponse.getStreamErrorsList()) {
-          System.out.println(err.getErrorMessage());
+    // One time initialization.
+    WritePendingStream writer = WritePendingStream.create(parentTable, client);
+
+    try {
+      // Write two batches to the stream, each with 10 JSON records.
+      for (int i = 0; i < 2; i++) {
+        // Create a JSON object that is compatible with the table schema.
+        JSONArray jsonArr = new JSONArray();
+        for (int j = 0; j < 10; j++) {
+          JSONObject record = new JSONObject();
+          record.put("col1", String.format("batch-record %03d-%03d", i, j));
+          jsonArr.put(record);
         }
-        throw new RuntimeException("Error committing the streams");
+        writer.append(jsonArr);
       }
-      System.out.println("Appended and committed records successfully.");
     } catch (ExecutionException e) {
       // If the wrapped exception is a StatusRuntimeException, check the state of the operation.
       // If the state is INTERNAL, CANCELLED, or ABORTED, you can retry. For more information, see:
       // https://grpc.github.io/grpc-java/javadoc/io/grpc/StatusRuntimeException.html
-      System.out.println(e);
+      System.out.println("Failed to append records. \n" + e);
+    }
+
+    // Final cleanup for the stream.
+    writer.cleanup();
+    System.out.println("Appended records successfully.");
+
+    // Once all streams are done, commit all of them in one request. This example only has the one
+    // stream.
+    BatchCommitWriteStreamsRequest commitRequest =
```

Comment:
We shouldn't commit for failed writes?

Reply:
Added an exception throw in cleanup and a comment about only committing successful streams.